watertap/flowsheets/crystallization/crystallizer_live_steam_with_condenser_chiller.py

```diff
@@ -0,0 +1,508 @@
+# of Energy). All rights reserved.
+#
+# Please see the files COPYRIGHT.md and LICENSE.md for full copyright and license
+# information, respectively. These files are also available online at the URL
+# "https://github.com/watertap-org/watertap/"
+#################################################################################
+from pyomo.environ import (
+    ConcreteModel,
+    TerminationCondition,
+)
+from pyomo.environ import (
+    ConcreteModel,
+    value,
+    Constraint,
+    Objective,
+    Var,
+    NonNegativeReals,
+    TransformationFactory,
+    units as pyunits,
+    check_optimal_termination,
+)
+from pyomo.network import Arc, SequentialDecomposition
+from idaes.core import FlowsheetBlock
+from watertap.core.solvers import get_solver
+from idaes.core.util.model_statistics import degrees_of_freedom
+from idaes.core.util.initialization import (
+    propagate_state,
+)
+from pyomo.util.check_units import assert_units_consistent
+
+from idaes.core import FlowsheetBlock
+from idaes.core.util.model_statistics import degrees_of_freedom
+
+import idaes.core.util.scaling as iscale
+import idaes.logger as idaeslog
+from watertap.core.solvers import get_solver
+from idaes.core import UnitModelCostingBlock
+
+from watertap.property_models.unit_specific import cryst_prop_pack as props
+from watertap.unit_models.crystallizer import Crystallization
+from watertap.costing import WaterTAPCosting, CrystallizerCostType
+from watertap.costing.unit_models.heat_exchanger import (
+    cost_heat_exchanger,
+)
+
+from io import StringIO
+from pyomo.util.infeasible import (
+    log_infeasible_constraints,
+)
+from watertap.unit_models.steam_heater_0D import SteamHeater0D, Mode
+from idaes.models.unit_models import Heater, Separator, Mixer, Product, Feed
+from idaes.models.unit_models.separator import SplittingType
+from watertap.unit_models.pressure_changer import Pump
+from watertap.costing.unit_models.pump import (
+    cost_pump,
+)
+from idaes.models.unit_models.mixer import MomentumMixingType
+from watertap.core.solvers import get_solver
+from watertap.unit_models.mvc.components.lmtd_chen_callback import (
+    delta_temperature_chen_callback,
+)
+from idaes.models.unit_models.heat_exchanger import (
+    HeatExchangerFlowPattern,
+)
+from watertap.costing.unit_models.heater_chiller import (
+    cost_heater_chiller,
+)
+from watertap.property_models.unit_specific import cryst_prop_pack as props
+import watertap.property_models.water_prop_pack as props_w
+import watertap.property_models.NaCl_T_dep_prop_pack as props_nacl
+from pyomo.common.log import LoggingIntercept
+from idaes.models.unit_models.translator import Translator
+import logging
+from watertap.core.util.model_debug_mode import activate
+activate()
+
+
+
+__author__ = "Elmira Shamlou"
+
+
+def main():
+    solver = get_solver()
+    m = build()
+    set_operating_conditions(m)
+    initialize_system(m, solver=solver)
+
+    optimize_set_up(m)
+
+    #interval_initializer(m)
```

Comment:
I am guessing that an InfeasibleConstraint exception (or something like that) arises when using the interval initializer.